src/net/net.c

```diff
@@ -196,11 +196,16 @@ static boolean get_static_config(tuple root, struct netif *n, boolean trace) {
     ip4_addr_t netmask;
     ip4_addr_t gw;
 
-    if (!get_config_addr(root, sym(ipaddr), &ip) ||
-        !get_config_addr(root, sym(gateway), &gw) ||
-        !get_config_addr(root, sym(netmask), &netmask))
+    if (!get_config_addr(root, sym(ipaddr), &ip))
         return false;
 
+    if (!get_config_addr(root, sym(gateway), &gw))
+        // assumes the default gateway is the first or last IP in the network range
+        gw.addr = (ip.addr & ~0xff) | ((ip.addr & 0xff) == 1 ? 254 : 1);
+
+    if (!get_config_addr(root, sym(netmask), &netmask))
+        netmask.addr = 4294967040; // 255.255.255.0
```

Comment:
Hex would be more readable here, and again use lwip_htonl. It may also be preferable to set addresses using "ip4_addr_set_u32" should the fields in ip4_addr_t change in a future lwIP release.

Reply:
Converted to hex.